admyral/compiler/action_parser.py

```diff
@@ -47,18 +47,26 @@ def parse_action(module_str: str, action_name: str) -> PythonAction:
     }
 
     # extract the action decorator arguments
-    if (
-        len(decorator_list) != 1
-        or (
-            isinstance(decorator_list[0], ast.Call)
-            and decorator_list[0].func.id != "action"
-        )
-        or (
-            isinstance(decorator_list[0], ast.Name) and decorator_list[0].id != "action"
-        )
-    ):
-        raise ValueError("Action function should only have the @action decorator.")
-    action_decorator = decorator_list[0]
+    action_decorator = None
+    for decorator in decorator_list:
+        if isinstance(decorator, ast.Call) and decorator.func.id == "action":
+            action_decorator = decorator
+            break
+        elif isinstance(decorator, ast.Name) and decorator.id == "action":
+            action_decorator = decorator
+            break
+
+    if action_decorator is None:
+        raise ValueError("Action function must have the @action decorator.")
+
+    if isinstance(action_decorator, ast.Call):
+        # Ensure the decorator has arguments if expected
+        if not action_decorator.keywords:
+            raise ValueError("The @action decorator must include arguments.")
+
+        for arg in action_decorator.keywords:
+            if not isinstance(arg.value, (ast.Constant, ast.List)):
+                raise ValueError(f"Invalid argument type in @action decorator: {arg.arg} must be a constant or list.")
```

Comment:
Good idea to validate the values. However, I would suggest a different approach due to maintainability and granularity reasons. It would probably make sense to add a parameter value validation function in `admyral/action.py` which is then used before a new action is constructed in the decorator and also here. If we maintain this function in `action.py` and a new parameter is added then it is harder to miss updating the validation function.